physical/raft/raft_autopilot.go

```diff
@@ -263,30 +268,37 @@ type Delegate struct {
 	*RaftBackend
 
 	// dl is a lock dedicated for guarding delegate's fields
-	dl               sync.RWMutex
-	inflightRemovals map[raft.ServerID]bool
-	emptyVersionLogs map[raft.ServerID]struct{}
+	dl                 sync.RWMutex
+	inflightRemovals   map[raft.ServerID]bool
+	emptyVersionLogs   map[raft.ServerID]struct{}
+	permanentNonVoters map[raft.ServerID]bool
 }
 
 func NewDelegate(b *RaftBackend) *Delegate {
 	return &Delegate{
-		RaftBackend:      b,
-		inflightRemovals: make(map[raft.ServerID]bool),
-		emptyVersionLogs: make(map[raft.ServerID]struct{}),
+		RaftBackend:        b,
+		inflightRemovals:   make(map[raft.ServerID]bool),
+		emptyVersionLogs:   make(map[raft.ServerID]struct{}),
+		permanentNonVoters: make(map[raft.ServerID]bool),
 	}
 }
 
 // AutopilotConfig is called by the autopilot library to know the desired
 // autopilot configuration.
 func (d *Delegate) AutopilotConfig() *autopilot.Config {
+	// Always fetch non-voters before returning the config
+	err := d.FetchNonVoters()
+	if err != nil {
+		d.logger.Error("failed to fetch non-voters", "error", err)
+	}
 	d.l.RLock()
 	config := &autopilot.Config{
 		CleanupDeadServers:      d.autopilotConfig.CleanupDeadServers,
 		LastContactThreshold:    d.autopilotConfig.LastContactThreshold,
 		MaxTrailingLogs:         d.autopilotConfig.MaxTrailingLogs,
 		MinQuorum:               d.autopilotConfig.MinQuorum,
 		ServerStabilizationTime: d.autopilotConfig.ServerStabilizationTime,
-		Ext:                     nil,
+		Ext:                     d.permanentNonVoters,
```

Comment:
I guess it doesn't truly matter since this value isn't persisted, but would we want to preemptively set this to a struct of our own and not directly place the non-voters map here in case we want to add more fields?

Reply:
This value is only used in the `CustomPromoter` struct. If we ever want to add more fields here in the future, I think this will be very straightforward to extend